doc/setup/native_libs.md

```diff
@@ -14,47 +14,116 @@ The key element for calling the GDAL native library from mapfishapp is the **ima
+
+Notice : this method doesn't allow to have multiple apps with gdal in the same tomcat, but actually we can't get shared classloader to work because gdal load the native and imageio also load them, there is a conflict.
+
+And finally edit your `/etc/default/tomcat8` file to have something like :
+```
+LD_LIBRARY_PATH="/usr/local/lib:${LD_LIBRARY_PATH}"
+JAVA_OPTS="${JAVA_OPTS} -Djava.library.path=$LD_LIBRARY_PATH"
+```
+
+### Install GDAL from sources
+
+If you do not want to use the precompiled binaries, another way to install the GDAL Java binding is building it from sources.
+
+The advantage to compile GDAL is to allow more formats to be supported (ecw, jp2K, ...), you can choose the ones which suits your needs.
+
+In order to compile we'll need some compiler packages :
+```
+apt-get install build-essential
+```
 
-Some more work is needed for installing the GDAL Java binding library, as there is still no deb package for it (note that packages exist for ruby and perl bindings, hopefully the Java's one will be released soon - see a recent proposal http://ftp-master.debian.org/new/gdal_1.10.0-0%7Eexp3.html).
+#### Enable ECW support
 
-To quickly install the GDAL Java binding library on the server, download and extract the library and its data. To do so, you can use 2 repositories providing binary packages for given distribution:
+For example, in order to enable the support of ECW format, we need to install it. We will explain how to install the version 2-3.3 of the ECW SDK, but this isn't the newer version and you can choose one more recent if you need but remember there are limits for the uses of this SDK with no licence.
 
- * GeoSolutions http://demo.geo-solutions.it/share/github/imageio-ext/releases/1.1.X/1.1.7/native/gdal/ which provides a package with ECW support (which can cause some licensing issues, and some of the packages are outdated, since the distributions evolved and updated their Glibc version, see #409) ;
- * geOrchestra-provided packages http://sdi.georchestra.org/~pmauduit/gdalogr-java-bindings/ which provides "vanilla" GDAL packages as well as a specific one (mifmid-patched) which allows the use of MIF/MID format across GDAL/OGR via GeoTools (see #409)
+First search `libecwj2-3.3-2006-09-06.zip` on Google in order to found the SDK. (no link because it always change)
 
+After you get this zip simply, we'll need to apply some patch to correct the jp2K support and som other bugs and then compile the sources and install it : 
+```
+unzip libecwj2-3.3-2006-09-06.zip
+rm libecwj2-3.3-2006-09-06.zip
+wget http://trac.osgeo.org/gdal/raw-attachment/ticket/3162/libecwj2-3.3-msvc90-fixes.patch
+patch -p0 < libecwj2-3.3-msvc90-fixes.patch
+wget http://osgeo-org.1560.x6.nabble.com/attachment/5001530/0/libecwj2-3.3-wcharfix.patch
+wget http://trac.osgeo.org/gdal/raw-attachment/ticket/3366/libecwj2-3.3-NCSPhysicalMemorySize-Linux.patch
```

Comment:
I suggest you create a gist with those files, and mention the gist URLs rather than those